lib/index.js

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2016 Digital Bazaar, Inc. All rights reserved.
+ */
+/* global webpage, window, document */
+var bedrock = require('bedrock');
+var cache = require('bedrock-redis');
+var config = bedrock.config;
+var express = require('bedrock-express').express;
+var phridge = require('phridge');
+var prerender = require('prerender-node');
+var querystring = require('querystring');
+var url = require('url');
+
+require('bedrock-requirejs');
+require('bedrock-views');
+
+var api = {};
+module.exports = api;
+
+require('./config');
+
+var logger = bedrock.loggers.get('app');
+var phantom = null;
+var requestQueue = {};
+
+var args = {
+  '--load-images': false,
+  '--ignore-ssl-errors': true,
+  '--ssl-protocol': 'tlsv1.2'
+};
+
+bedrock.events.on('bedrock.start', callback => {
+  // configure whitelist / blacklist
+  if(config.prerender.whitelist) {
+    prerender.whitelisted(config.prerender.whitelist);
+  }
+  if(config.prerender.blacklist) {
+    prerender.blacklisted(config.prerender.blacklist);
+  }
+  phridge.spawn(args).then(p => {
+    phantom = p;
+    return phantom.run(function() {
+      console.log('PhantomJS Started.');
+    });
+  }).then(() => callback(), callback);
+});
+
+bedrock.events.on('bedrock-express.configure.routes', (app, callback) => {
+  var routes = config.prerender.routes;
+  var routeKeys = Object.keys(config.prerender.routes);
+  if(routeKeys.length) {
+    var router = express.Router();
+    var error;
+    var success = routeKeys.every(route => {
+      if(typeof routes[route] !== 'object') {
+        error = new TypeError(
+          'Route options must be an object. Route: \'' + route +
+          '\' Value: \'' + routes[route] + '\'');
+        return false;
+      }
+      router.get(route, _prerender);
+      return true;
+    });
+    if(!success) {
+      return callback(error);
+    }
+    app.use(router);
+  }
+  callback();
+});
+
+function _prerender(req, res, next) {
+  if(!prerender.shouldShowPrerenderedPage(req)) {
+    return next();
```

Comment:
Yeah, but if we're in the `_prerender` function ... there's no other middleware after it, right? Seems like you'd always get a 404. So, I'm a little confused as to how this could even work if it doesn't move onto another route. It would be good to know exactly where it goes when it hits that path.